File: server/src/main/java/com/mthwate/conk/block/BlockBase.java
package com.mthwate.conk.block;

/**
 * @author mthwate
 */
public abstract class BlockBase implements Block {

	protected Data data = new Data();

	@Override
	public void setData(Data data) {
		this.data = data;
	}

	@Override
	public Data getData() {
		return data;
	}

	@Override
	public boolean isSolid() {
		return true;
	}

	@Override
	public boolean isTransparent() {
		return false;
	}

	@Override
	public String[] getTextures() {
		return new String[] {getName()};
	}

}

File: server/src/main/kotlin/com/mthwate/conk/block/BlockBase.kt
package com.mthwate.conk.block

/**
 * @author mthwate
 */
abstract class BlockBase(override val name: String) : Block {

	override val isSolid: Boolean
		get() = true

	override val isTransparent: Boolean
		get() = false

	override val textures: Array<String>
		get() = arrayOf(name)

}
